src/deadline/job_attachments/download.py

```diff
@@ -908,6 +1017,38 @@ def merge_asset_manifests(manifests: list[BaseAssetManifest]) -> BaseAssetManife
     return output_manifest
 
 
+def merge_asset_manifests_sorted_asc_by_last_modified(
```

Comment:
Do we need this function? What this does will be simple to read and understand in any client code calling it. Maybe prefix it with `_` to keep it internal for now, so that we can decide that later.